pkg/cloudprovider/aws/instancetypes_test.go

```diff
@@ -0,0 +1,259 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws_test
```

Comment:
I'm a bit wary of this two packages one folder model. Any reason this can't just be in the aws package? In general, we've also been following more of a e2e test pattern instead of unit tests since this is so heavily mocked. Right now it kinda feels like we're programming the EC2 mocks and then asserting that the mocks behave as expected. Maybe worth an offline call to chat through testing? I'm not necessarily opposed to any of this, I just want to make sure we're aligned on the purpose.

Reply:
Let's definitely chat about this.  I like the secondary package for tests since it makes you think about the interface to the package rather than relying on all the private members. There is a lot of mock action here, but I still like having the unit tests since there is still outside logic that manipulates the data from the API. Especially here for testing the AZ filtering (since it comes from InstanceTypeOfferings) and the resource filters.  It definitely doesn't replace the e2e tests, they're probably more important.